server/server.go

```diff
@@ -1,154 +0,0 @@
-package main
-
-import (
-	"fmt"
-	"io/ioutil"
-	"log"
-	"strconv"
-	"net/http"
-	"encoding/json"
-	"database/sql"
-
-	"github.com/willf/bloom"
-	"github.com/blics18/SendGrid/client"
-	_ "github.com/go-sql-driver/mysql"
-	
-)
-
-type bloomFilter struct {
-	filter *bloom.BloomFilter
-}
-
-func createBloomFilter() *bloomFilter {
-	return &bloomFilter{
-		filter: bloom.New(20*uint(1000), 5),
-	}
-}
-
-func (bf *bloomFilter) populateBF(w http.ResponseWriter, r *http.Request) {
-	if r.Body == nil {
-		w.Write([]byte("Could not read the body of the request"))
-	}
-
-	body, err := ioutil.ReadAll(r.Body)
-	if err != nil {
-		w.WriteHeader(http.StatusBadRequest)
-		w.Write([]byte("Could not read the body of the request"))
-	}
-
-	defer r.Body.Close()
-
-	var users []client.User
-
-	err = json.Unmarshal(body, &users)
-	if err != nil {
-		w.WriteHeader(http.StatusBadRequest)
-		w.Write([]byte("Unable to parse json body"))
-		return
-	}
-
-	w.WriteHeader(http.StatusOK)
-	w.Write([]byte(strconv.Itoa(http.StatusOK)))
-
-	for _, user := range users {
-		for _, email := range user.Email {
-			bf.filter.Add([]byte(fmt.Sprintf("%d|%s", *user.UserID, email)))
-			fmt.Println(fmt.Sprintf("userID: %d", *user.UserID))
-			fmt.Println(fmt.Sprintf("Email: %s", email))
-		}
-	}
-}
-
-func (bf *bloomFilter) checkBF(w http.ResponseWriter, r *http.Request) {
-	if r.Body == nil {
-		w.WriteHeader(http.StatusBadRequest)
-		w.Write([]byte("Need content to check"))
-		return
-	}
-
-	body, err := ioutil.ReadAll(r.Body)
-	if err != nil {
-		w.WriteHeader(http.StatusBadRequest)
-		w.Write([]byte("Could not read the body of the request"))
-	}
-
-	defer r.Body.Close()
-
-	var user client.User
-
-	err = json.Unmarshal(body, &user)
-	if err != nil {
-		w.WriteHeader(http.StatusBadRequest)
-		w.Write([]byte("Unable to parse json body"))
-		return
-	}
-
-	if &user.UserID == nil {
-		w.WriteHeader(http.StatusBadRequest)
-		w.Write([]byte("Need User ID"))
-		return
-	}
-
-	if user.Email == nil {
-		w.WriteHeader(http.StatusBadRequest)
-		w.Write([]byte("Need User Emails"))
-		return
-	}
-
-	for _, email := range user.Email {
-		if bf.filter.Test([]byte(fmt.Sprintf("%d|%s", *user.UserID, email))) {
-			w.Write([]byte(email + " is in the bloom filter. Cross checking..."))
-			if crossCheck(user.UserID, email) {
-				w.Write([]byte(email + " is in the database"))
-				fmt.Println(email + " is in the database")
-			} else {
-				w.Write([]byte(email + " is not in the database"))
-				fmt.Println(email + " is not in the database")
-			}
-		} else {
-			w.Write([]byte(email + " is not in the bloom filter"))
-		}
-	}
-}
-
-func crossCheck(UserID *int, Email string) bool {

```

Comment:
This function needs to bubble up the errors it encounters. Consider changing this functions return signature
```
func crossCheck(UserID *int, Email string) (bool, error) {...}
```

This way if there is an error (e.g. with the database connection) then this would return an error, instead of `false` which may or may not be the case.